samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -743,7 +745,7 @@ static int sock_send(
 		}
 	}
 
-	bytes = send(socket_info->fd, data, length, 0);
+	bytes = send(socket_info->fd, data, length, flags);
```

Comment:
I'm thinking if setting the blocking mode could be handled here but haven't really been able to think through all possible scenarios.

Reply:
Maybe this is not the most optimal place as we call this in a loop in `sock_send_random_data_length()`.

Reply:
This comment is outdated as I've understood few things incorrectly at the beginning.

Reply:
> I'm thinking if setting the blocking mode could be handled here but haven't really been able to think through all possible scenarios.

Actually I think it's a good idea to use `MSG_DONTWAIT` here instead of `fcntl(fd, F_SETFL, O_NONBLOCK)`. When sending a large random data block the `recv()` may happen during the `send()` loop, and in this case the `recv()` will be either blocking or non-blocking, depending on if the `send()` loop is done or not. I observe this when testing with `sock send -i0 -l65535`.